src/slice.c

```diff
@@ -475,8 +475,18 @@ SEXP vec_init(SEXP x, R_len_t n) {
 
 // [[ register() ]]
 SEXP vctrs_init(SEXP x, SEXP n) {
+  n = PROTECT(vec_cast(n, vctrs_shared_empty_int, args_n, args_empty));
+
+  if (r_length(n) != 1) {
+    r_abort("`n` must have size 1, not size %i.", r_length(n));
```

Comment:
Perhaps move this to a draft `vec_assert()` function (even if it only handles `size`)? This way there is less risk to forget to update this code once we have an actual C API for `vec_assert()`.